compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -113,6 +113,49 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 M::retag(self, *kind, dest)?;
             }
 
+            // Call CopyNonOverlapping
+            CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping { dst, src, count }) => {
+                let (src, size) = {
+                    let src = self.eval_operand(src, None)?;
+                    let size = src.layout.layout.size;
+                    let ptr = match src.try_as_mplace(self) {
+                        Ok(mplace) => match mplace.ptr {
+                            Scalar::Ptr(ptr) => ptr,
+                            _ => panic!(),
+                        },
+                        Err(imm) => match self.ref_to_mplace(imm)?.ptr {
+                            Scalar::Ptr(ptr) => ptr,
+                            _ => panic!(),
+                        },
+                    };
+                    (ptr, size)
+                };
+
+                let dst = {
+                    let dst = self.eval_operand(dst, None)?;
+                    match dst.try_as_mplace(self) {
+                        Ok(mplace) => match mplace.ptr {
+                            Scalar::Ptr(ptr) => ptr,
+                            _ => panic!(),
+                        },
+                        Err(imm) => match self.ref_to_mplace(imm)?.ptr {
+                            Scalar::Ptr(ptr) => ptr,
+                            _ => panic!(),
+                        },
+                    }
+                };
+
+                let count = self.eval_operand(count, None)?;
+                let count = self.read_immediate(count)?.to_scalar()?;
+                let count = if let Scalar::Int(i) = count {
+                    core::convert::TryFrom::try_from(i).unwrap()
+                } else {
+                    panic!();
+                };
```

Comment:
I think this is possible:

```suggestion
                let count = self.read_immediate(count)?.to_scalar()?.to_u64()?
```